crates/aranya-fast-channels/src/client.rs

```diff
@@ -102,7 +102,11 @@ impl<S: AfcState> Client<S> {
         data.try_reserve_exact(Self::OVERHEAD)?;
 
         // Append zeros for the tag and header.
-        data.try_resize(data.len() + Self::OVERHEAD, 0)?;
+        let new_len = data
+            .len()
+            .checked_add(Self::OVERHEAD)
+            .ok_or_else(|| Self::unlikely(Error::InputTooLarge))?;
```

Comment:
This should be `.assume` because if `checked_add` returns `None` then `try_reserve_exact` should've already returned an error.